src/widgets/helper/ChannelView.cpp

```diff
@@ -1512,15 +1518,30 @@ void ChannelView::mouseMoveEvent(QMouseEvent *event)
         this->currentMousePosition_ = event->screenPos();
     }
 
-    // is selecting
+    // check for word underneath cursor
+    const MessageLayoutElement *hoverLayoutElement =
+        layout->getElementAt(relativePos);
+
+    //selecting single characters
     if (this->isLeftMouseDown_)
     {
         auto index = layout->getSelectionIndex(relativePos);
-
         this->setSelection(this->selection_.start,
                            SelectionItem(messageIndex, index));
     }
 
+    //selecting whole words
+    if (this->isDoubleClick_ && hoverLayoutElement)
+    {
+        auto [wordStart, wordEnd] =
+            this->getWordBounds(layout.get(), hoverLayoutElement, relativePos);
+        auto wordSelection = Selection{SelectionItem(messageIndex, wordStart),
+                                       SelectionItem(messageIndex, wordEnd)};
+        auto selectUnion = this->doubleClickSelection_ |= wordSelection;
```

Comment:
```suggestion
        auto selectUnion = this->doubleClickSelection_ | wordSelection;
```

Maybe a comment here would be helpful, because `doubleClickSelection_` never changes when moving the mouse (and no messages are added). Something along the lines of _"doubleClickSelection marks the initially selected word. selectUnion spans the selection from the initial word to the current word."_.

Reply:
added comment and renamed wordSelection to show intent better 3126d1dc2aa561075cd96545e52218ca06a0f384